src/components/icon-button/icon-button.pw.tsx

```diff
@@ -0,0 +1,227 @@
+import React from "react";
+import { test, expect } from "@playwright/experimental-ct-react17";
+import IconButtonComponent from "./component.test-pw";
+import { CHARACTERS } from "../../../playwright/support/constants";
+import { HooksConfig } from "../../../playwright";
+import { checkAccessibility } from "../../../playwright/support/helper";
+
+test.describe("when focused", () => {
+  test(`should have the expected styling when the focusRedesignOptOut is false`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<IconButtonComponent aria-label={CHARACTERS.STANDARD} />);
+
+    const iconLocator = page.getByTestId("icon").locator("..");
+    const element = await iconLocator;
+    await element.focus();
+    await expect(element).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+    await expect(element).toHaveCSS("outline", "rgba(0, 0, 0, 0) solid 3px");
+  });
+
+  test(`should have the expected styling when the focusRedesignOptOut is true`, async ({
+    mount,
+    page,
+  }) => {
+    await mount<HooksConfig>(
+      <IconButtonComponent aria-label={CHARACTERS.STANDARD} />,
+      {
+        hooksConfig: { focusRedesignOptOut: true },
+      }
+    );
+
+    const iconLocator = page.getByTestId("icon").locator("..");
+    const element = await iconLocator;
+    await element.focus();
+    await expect(element).toHaveCSS("outline", "rgb(255, 188, 25) solid 3px");
+  });
+});
+
+test.describe("check props for IconButton component", () => {
+  test(`should render IconButton with aria-label prop`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<IconButtonComponent aria-label={CHARACTERS.STANDARD} />);
+
+    const iconLocator = page.getByTestId("icon").locator("..");
+
+    await expect(iconLocator).toHaveAttribute(
+      "aria-label",
+      CHARACTERS.STANDARD
+    );
+    await iconLocator.isVisible();
+  });
+
+  test(`should render IconButton with children prop`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<IconButtonComponent />);
+
+    const iconLocator = page.getByTestId("icon");
+    await iconLocator.isVisible();
+  });
+
+  test(`should render IconButton with disabled prop`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<IconButtonComponent disabled />);
+
+    const iconLocator = page.getByTestId("icon").locator("..");
+    await iconLocator.isDisabled();
+    await iconLocator.getAttribute("disabled");
```

Comment:
This would be preferable:
`await expect(iconLocator).toBeDisabled()`
`await expect(iconLocator).toHaveAttribute("disabled", /.*/);`